File: book/src/schema/enums/advanced.go
package main

func main() {}

type SampleVariant interface {
	sealed()
}

type Sample SampleVariant

type Sample_Variant1 struct {
	F1 uint32
	F2 uint16
}

func (v Sample_Variant1) sealed() {}

type Sample_Variant2 struct {
	Field1 int64
	Field2 bool
}

func (v Sample_Variant2) sealed() {}

// N variants...

File: book/src/reference/schema/enums/advanced.py
from dataclasses import dataclass
from typing import Literal


# region snippet
@dataclass
class SampleVariant1:
    tag: Literal["Variant1"]
    f1: int
    f2: int


@dataclass
class SampleVariant2:
    tag: Literal["Variant2"]
    field1: int
    field2: bool


# endregion snippet


Sample = SampleVariant1 | SampleVariant2
